schema/vrs.yaml

```diff
@@ -396,27 +387,68 @@ definitions:
     required: [ "type", "gene_id" ]
 
   # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-  # Range
+  # Numerics, Comparators, and Ranges
 
   Range:
     description: >-
-      An inclusive range of integers. A range may be unbounded by omitting either
-      min or max values. Fully unbounded ranges are not permitted (one of min or
-      max MUST be specified).
+      A set of classes that provide the possible representations of integer number
+      ranges, from a single integer (BoundedInteger) to bounded ranges (ClosedRange)
+      or half-bounded ranges (OpenRange).
+    oneOf:
+      - $ref: "#/definitions/BoundedInteger"
+      - $ref: "#/definitions/BoundedRange"
+      - $ref: "#/definitions/HalfBoundedRange"
+    discriminator:
+      propertyName: type
+
+  BoundedInteger:
+    description: >-
+        A range bounding a single integer.
+    type: object
+    additionalProperties: false
+    properties:
+      type:
+        type: string
+        enum: [ "BoundedInteger" ]
+        default: "BoundedInteger"
+      value:
+        type: integer
+    required: ["type", "value"]
+
+  HalfBoundedRange:
+    description: >-
+      A half-open range with an open bound indicated by comparator. The bound operator
+      is interpreted as follows: '>=' are all integers greater than and including
+      the value, '<=' are all integers less than and including the value.
+    type: object
+    additionalProperties: false
+    properties:
+      type:
+        type: string
+        enum: [ "HalfBoundedRange" ]
+        default: "HalfBoundedRange"
+      bound:
```

Comment:
I think this should be called "value" as well (not bound). Since it is the "value" that is known and will almost certainly be used in similar fashions as the value in the BoundedInteger but with some additional considerations. We may want to consider carrying the "value" attribute thru to the BoundedRange below and simply add an "outerValue" so that we can consider the "value a consistent basis for each of the Integer subtypes and the comparator (in the open range) and the outerValue  in the (bounded range) would be the extensions that these additional "range" classes use to provide the level of ambiguity commensurate with the requirements we've identified.

Value is an attribute that should be on the ancestor class. And all descendants should be able to define themselves relative to a base-value. If not, then they should be considered different classes. This is why the BoundedRange (IMO) needs to either be a separate class which references SimpleInteger (or BoundedInteger above) or it should have the notion of it's base-value and added attributes to represent the "other" limit or bound that satisfies our requirements for Counts and Positions.

Reply:
Are you trying to cater for uncertain ranges? Like the `STARTCI` and `ENDCI` in the VCF spec? e.g. trying to say "I know it starts  at position 500, -200, + 300 and ends at about 3000, -450, + 550". The `HalfBoundedRange` '`>=10`' is equivalent to `[10, infinity]` and '`<=10`' = `[1, 10]`, assuming you're using an unsigned, one-based coordinate system.

Reply:
> I think this should be called "value" as well (not bound). Since it is the "value" that is known and will almost certainly be used in similar fashions as the value in the BoundedInteger but with some additional considerations. We may want to consider carrying the "value" attribute thru to the BoundedRange below and simply add an "outerValue" so that we can consider the "value a consistent basis for each of the Integer subtypes and the comparator (in the open range) and the outerValue in the (bounded range) would be the extensions that these additional "range" classes use to provide the level of ambiguity commensurate with the requirements we've identified.

No problem. I will change this and similar attributes back to `value`.

> Value is an attribute that should be on the ancestor class. And all descendants should be able to define themselves relative to a base-value. If not, then they should be considered different classes. This is why the BoundedRange (IMO) needs to either be a separate class which references SimpleInteger (or BoundedInteger above) or it should have the notion of it's base-value and added attributes to represent the "other" limit or bound that satisfies our requirements for Counts and Positions.

Okay, yes, I agree. I will make these changes.